internal/controller/api/v1alpha1/nmstate_config_types.go

```diff
@@ -20,19 +20,19 @@ import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
-type RawConfig []byte
+type MacInterfaceMapItems struct {
+	// nic name used in the yaml, which relates 1:1 to the mac address
+	LogicalNicName string `json:"logicalNICName"`
+	// mac address present on the host.
+	// Pattern: ^([0-9A-Fa-f]{2}[:]){5}([0-9A-Fa-f]{2})$
+	MacAddress string `json:"macAddress"`
+}
 
 type NMStateConfigSpec struct {
-	// MACAddress is a MAC address for any network device on the host to which this config
-	// should be applied. This value is only used to ensure that the config is applied to the
-	// intended host.
-	MACAddress string `json:"macAddress"`
-
-	//Config contains the  yaml [1] as string instead of golang struct so we don't need to be in
-	//sync with the schema.
-	//
-	// [1] https://github.com/nmstate/nmstate/blob/base/libnmstate/schemas/operational-state.yaml
-	Config RawConfig `json:"-"`
+	// mapping of host macs to logical interfaces used in the network yaml
```

Comment:
We should expand on exactly how this will be used. Something like:

```
// Interfaces is an array of interface objects containing the name and MAC 
// address for interfaces that are referenced in the raw nmstate config YAML.
// Interfaces listed here will be automatically renamed in the nmstate config
// YAML to match the real device name that is observed to have the
// corresponding MAC address. At least one interface must be listed so that it
// can be used to identify the correct host, which is done by matching any MAC
// address in this list to any MAC address observed on the host.
```

Reply:
We can also use a validation marker to enforce a minimum of 1 item in the array.

Reply:
Done